src/Libraries/CoreNodes/List.cs

```diff
@@ -509,32 +509,41 @@ public static IList RestOfItems(IList list)
         }
 
         /// <summary>
-        ///     Chop a list into a set of lists each containing the given amount of items.
+        ///     Chop a list into a set of consecutive sublists with the specified lengths. List division begins at the top of the list.
         /// </summary>
-        /// <param name="list">List to chop up.</param>
-        /// <param name="subLength">Length of each new sub-list.</param>
-        /// <returns name="lists">List of lists.</returns>
-        /// <search>sublists,build sublists,slices,partitions,cut,listcontains</search>
-        public static IList Chop(IList list, int subLength)
+        /// <param name="list">List to chop into sublists</param>
+        /// <param name="lengths">Lengths of consecutive sublists to be created from the input list</param>
+        /// <returns name="lists">Sublists created from the list</returns>
+        /// <search>chop</search>
+        public static IList Chop(IList list, IList lengths)
         {
-            if (list.Count < subLength)
-                return list;
-
             var finalList = new ArrayList();
             var currList = new ArrayList();
             int count = 0;
+            int lengthIndex = 0;
 
-            foreach (object v in list)
+            // If there is no any lengths more than 0,
+            // we return empty list.
+            if (lengths.Cast<int>().All(x => x <= 0))
```

Comment:
hmm, this changes the behavior of the old list.chop node: 

![screen shot 2015-09-15 at 1 51 26 pm](https://cloud.githubusercontent.com/assets/508936/9884998/e070215e-5bb0-11e5-8c2e-77346b029774.png)

I'm not sure if we should change the behavior: it will break user workflows, it might break tests.... @lukechurch ? @mccrone